### PR TITLE
Move Janusgraph docker into main repo

### PR DESCRIPTION
**Pull request by farodin91**
## ToDo

* [ ] set latest tag
* [ ] set 1.0 or 0.6 tag

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

**Comment by farodin91** [comment]
Just tags are missing these can be added later.

File: janusgraph-dist/pom.xml [hunk 1]
```diff
@@ -582,6 +587,64 @@
                         </executions>
                     </plugin>
 
+                    <plugin>
```
Comment (FlorianHockmann) [inline janusgraph-dist/pom.xml:590]: This is part of the `janusgraph-release` profile which of course makes sense as that also builds the distribution archives. We should document that here: https://github.com/JanusGraph/janusgraph/blob/master/BUILDING.md#building-docker-image-for-janusgraph-server
and probably also directly above that in the sentence:
> This command generates the distribution archive [...]

It at least wasn't obvious to me directly that I had to enable that profile to build the docker image.

File: janusgraph-dist/docker/test-image.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+IMAGE=$1
+
+if [ -z "${IMAGE:-}" ]; then
```
Comment (FlorianHockmann) [inline janusgraph-dist/docker/test-image.sh:21]: This doesn't seem to work. I at least get an error if I don't specific an image-name:

```sh
$ janusgraph-dist/docker/test-image.sh
janusgraph-dist/docker/test-image.sh: line 19: $1: unbound variable
```

File: janusgraph-dist/docker/build-and-push-image.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (farodin91) [reply janusgraph-dist/docker/build-and-push-image.sh:1]: I didn't want to duplicate code.

File: janusgraph-dist/docker/test-image.sh (hunk 2 above)
Comment (farodin91) [reply janusgraph-dist/docker/test-image.sh:21]: fixed, i justed copied this file.

File: docs/operations/container.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+| Tag            | Support level | Docker base image      |
+|:--------------|:-------------|------------------------|
+| latest         | <ul><li>latest JanusGraph release</li><li>no breaking changes guarantees</li></ul> | eclipse-temurin:11-jre |
+| x.x            | <ul><li>newest patch-level version of JanusGraph</li><li>expect breaking changes</li></ul> | eclipse-temurin:8-jre  |
+| x.x.x          | <ul><li>defined JanusGraph version</li><li>breaking changes are only in this repo</li></ul> | eclipse-temurin:8-jre  |
+| x.x.x-revision | <ul><li>defined JanusGraph version</li><li>defined commit in JanusGraph repo</li></ul> | eclipse-temurin:8-jre  |
+
+
+[JG]: https://janusgraph.org/
+[JG_BDB]: https://docs.janusgraph.org/storage-backend/bdb/
```
[the remark this answers is not in the document]
Comment (farodin91) [reply docs/operations/container.md:308]: done

File: docs/operations/container.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+index.search.directory=/var/lib/janusgraph/index
+
+$ docker exec janusgraph-default grep threadPoolWorker /etc/opt/janusgraph/janusgraph-server.yaml
+threadPoolWorker: 2
+```
+
+##### Example: Cassandra-ES with Docker Compose
+
+Start a JanusGraph instance with Cassandra and Elasticsearch using the `cql-es`
+template through [`docker-compose-cql-es.yml`](docker-compose-cql-es.yml):
```
[the remark this answers is not in the document]
Comment (farodin91) [reply docs/operations/container.md:163]: i updated the links. I didn't to change that much in the docs in this PR.

File: docs/operations/container.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
+# JanusGraph Container
+
+This repository contains build scripts and configuration files for the official
```
[the remark this answers is not in the document]
Comment (farodin91) [reply docs/operations/container.md:3]: Good idea, we should do this in a upcoming PR to mix this process to much.

File: README.md [hunk 7]
```diff
@@ -13,6 +13,7 @@ can support thousands of concurrent users, complex traversals, and analytic grap
 [![Mentioned in Awesome Bigtable][awesome-shield]][awesome-link]
 [![CII Best Practices][bestpractices-shield]][bestpractices-link]
 [![Codacy Badge][codacy-shield]][codacy-link]
+[![Docker pulls][docker-pulls-img]][docker-hub-url]
```
[the remark this answers is not in the document]
Comment (farodin91) [reply README.md:16]: done

**Review by FlorianHockmann — APPROVED** [review]
I agree that it makes sense to improve the docs in a follow-up PR. I've created an issue to keep track of this #3793. I also added the image tags there as it sounded like you also wanted to do that in a different PR. Feel free to edit the issue if needed.

One last question though: Should we disable the deployment until the tagging is in place?

Apart from that, this PR looks good to me.

**Comment by farodin91** [comment]
@FlorianHockmann  I don't tagging is an issue, we are re-tag missing tags.

**Comment by farodin91** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|v0.6|[](https://github.com/JanusGraph/janusgraph/pull/3795)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**